apps/wallet-mobile/src/features/Swap/useCases/ConfirmTxScreen/ConfirmTxScreen.tsx

```diff
@@ -1,39 +1,139 @@
 
 export const ConfirmTxScreen = () => {
   const spendingPasswordRef = React.useRef<RNTextInput>(null)
-  const [confirmationModal, setConfirmationModal] = React.useState<boolean>(false)
-  const [bottomSheetState, setBottomSheetSate] = React.useState<{isOpen: boolean; title: string; content?: string}>({
-    isOpen: false,
-    title: '',
-    content: '',
-  })
-
+  const [screenState, setScreenState] = React.useState<
+    {modal: boolean; swapTx: undefined} | {modal: boolean; swapTx: YoroiUnsignedTx}
+  >({modal: false, swapTx: undefined})
+  const [orderDataFromHelper, setOrderDataFromHelper] = React.useState<Swap.CreateOrderData>()
   const [spendingPassword, setSpendingPassword] = React.useState('')
   const strings = useStrings()
   const wallet = useSelectedWallet()
 
-  const {createOrder} = useSwap()
-  const {selectedPool, amounts} = createOrder
+  const {createOrder: createOrderState} = useSwap()
+  const {selectedPool, amounts} = createOrderState
+
+  const {poolList} = usePoolsByPair({
+    tokenA: createOrderState.amounts.sell.tokenId,
+    tokenB: createOrderState.amounts.buy.tokenId,
+  })
+
   const buyTokenInfo = useTokenInfo({wallet, tokenId: amounts.buy.tokenId})
   const sellTokenInfo = useTokenInfo({wallet, tokenId: amounts.sell.tokenId})
   const tokenToBuyName = buyTokenInfo.ticker ?? buyTokenInfo.name
 
-  const calculatedFee = (Number(selectedPool?.fee) / 100) * Number(createOrder.amounts.sell.quantity)
+  const calculatedFee = (Number(selectedPool?.fee) / 100) * Number(createOrderState.amounts.sell.quantity)
   const poolFee = Quantities.format(`${calculatedFee}`, sellTokenInfo.decimals ?? 0)
 
+  const addresses = useAddresses()
+
+  // TODO create entry
+  /* 
+    Seliling ada:
+    {
+      address: addr
+      amounts:{
+        "":amount-sending+deposit pairpool
+      }
+    }
+
+    Selling anyting else:
+    {
+      address: addr
+      amounts:{
+        "":deposit pairpool
+        "tokenId":'4343434'
+      }
+    }    
+  
+*/
+  const createEntry = (): YoroiEntry => {
+    const amountEntry = {}
+    const tokenId = orderDataFromHelper?.amounts.sell.tokenId
+    if (tokenId != null) {
+      amountEntry[tokenId] = orderDataFromHelper?.amounts.sell.quantity
+    }
+
+    return {
+      address: orderDataFromHelper?.address !== undefined ? orderDataFromHelper.address : '',
+      amounts: amountEntry,
+    }
+  }
+
+  const {refetch} = useSwapTx(
+    // TODO add datum
+    {wallet, entry: createEntry()},
+    {
+      onSuccess: (yoroiUnsignedTx) => {
+        console.log('CREATE UNSIGNED TX SUCCESS: ', yoroiUnsignedTx)
+        setScreenState({...screenState, modal: true, swapTx: yoroiUnsignedTx})
+      },
+    },
+  )
+
+  const {createOrder} = useCreateOrder({
+    onSuccess: (data) => {
+      console.log('create order success', data)
+      // TODO: unsign TX
+      refetch()
+    },
+    onError: (error) => {
+      console.log('create order error', error)
+    },
+  })
+
+  useEffect(() => {
```

Comment:
create on the `onConfirm` so you don't need to use effect.

Reply:
How I added now is when the user enters the Confirm transaction screen the helpers (makePossibleMarketOrder or makeLimitOrder) run in the background and get the data for when the user presses confirm. I can move them as well to run them only when the user press confirm transaction btn